### DISCUSSION TITLE
(some sort of) api/bash tool for automatic creation of proxied host entries etc?

### DISCUSSION THREAD

**Discussion by posta246** [NginxProxyManager/nginx-proxy-manager#1548] [Ideas]
Hi,
I'm using a lot your fantastic tools, I'm questioning myself if there is - or could be, in the future - some kind of easy apis/tools/dontKnowWhat that allow me to create an entry via bash script.
I think I could doing it today simulating the http(s?) get/post that has been done by the "Save" button of the "Add Proxy Host" form.
Someone has done it before? Tips?
thanks!!

**Comment by chaptergy** [comment]
Please use the search before posting, there are already multiple issues / discussion about this. NPM uses an API internally which you can use, but it is not documented. Replicating the requests sent by the frontend is the easiest method to do this. See https://github.com/jc21/nginx-proxy-manager/issues/341#issuecomment-839129341.

**Comment by posta246** [comment]
wrote a sort of stub, here https://www.reddit.com/r/nginxproxymanager/comments/qmiw0q/linux_script_for_semiautomatic_proxy_hosts/
